internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,145 @@
+package olm
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	core "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+// RegistryPod holds resources necessary for creation of a registry server
+type RegistryPod struct {
+	registryImage       string
+	registryPodGRPCPort int32
+	registryDBPath      string
+	registryBundleImage string
```

Comment:
Export these fields (so they can be set by a consumer of this struct), remove the `registry` prefix since these are fields of a `RegistryPod` struct, and add a `BundleAddMode` field.
```suggestion
	IndexImage    string
	DBPath        string
	BundleImage   string
	BundleAddMode string
```